packages/store/src/legacyWeakMap.js

```diff
@@ -1,50 +1,40 @@
 // Copyright (C) 2019 Agoric, under Apache license 2.0
 
-// @ts-check
-
 import { assert, details as X, q } from '@agoric/assert';
-import { isEmptyNonRemotableObject } from './helpers.js';
 import './types.js';
 
 /**
- * @template {Record<any, any>} K
- * @template {any} V
- * @param {string} [keyName='key']
- * @returns {WeakStore<K, V>}
+ * @deprecated switch to ScalarWeakMap if possible, WeakMap otherwise
+ * @param {string} [keyName='key'] - the column name for the key
  */
-export function makeWeakStore(keyName = 'key') {
+export const makeLegacyWeakMap = (keyName = 'key') => {
   const wm = new WeakMap();
   const assertKeyDoesNotExist = key =>
     assert(!wm.has(key), X`${q(keyName)} already registered: ${key}`);
   const assertKeyExists = key =>
     assert(wm.has(key), X`${q(keyName)} not found: ${key}`);
-  const assertNotBadKey = key =>
-    assert(!isEmptyNonRemotableObject(key), X`${q(keyName)} bad key: ${key}`);
-  return harden({
+  const legacyWeakMap = {
     has: key => {
-      assertNotBadKey(key);
+      // .has is very accepting
```

Comment:
Can we explain more here too? Do you mean that we don't assert `!isEmptyNonRemotableObject` anymore?

Reply:
We no longer need `isEmptyNonRemotableObject`. There is no longer anything special about empty objects. A frozen empty object is just another copyRecord.